eng/common/TestResources/New-TestResources.ps1

```diff
@@ -295,16 +309,21 @@ foreach ($templateFile in $templateFiles) {
 
         if ($variable.Type -eq 'String' -or $variable.Type -eq 'SecureString') {
             $deploymentOutputs[$key] = $variable.Value
+        }
+    }
 
-            if ($CI) {
-                # Treat all ARM template output variables as secrets since "SecureString" variables do not set values.
-                # In order to mask secrets but set environment variables for any given ARM template, we set variables twice as shown below.
-                Write-Host "Setting variable '$key': ***"
-                Write-Host "##vso[task.setvariable variable=_$key;issecret=true;]$($variable.Value)"
-                Write-Host "##vso[task.setvariable variable=$key;]$($variable.Value)"
-            } else {
-                Write-Host ($shellExportFormat -f $key, $variable.Value)
-            }
+    foreach ($key in $deploymentOutputs.Keys)
+    {
+        $value = $deploymentOutputs[$key]
+        
+        if ($CI) {
+            # Treat all ARM template output variables as secrets since "SecureString" variables do not set values.
+            # In order to mask secrets but set environment variables for any given ARM template, we set variables twice as shown below.
+            Write-Host "Setting variable '$key': ***"
+            Write-Host "##vso[task.setvariable variable=_$key;issecret=true;]$($value)"
+            Write-Host "##vso[task.setvariable variable=$key;]$($value)"
+        } else {
+            Write-Host ($shellExportFormat -f $key, $value)
```

Comment:
Because of this line, it's going to dump a bunch of variables you defined that aren't actually used today. Some of these are already scoped to the service in the template. I don't think that large block above should be added at this time until a need arises.

Reply:
The need is arising, we are onboarding track 2 management libraries, all of them need AAD, subscription id, resource group etc.

Reply:
Why do they need different SPs? The instructions to create an SP don't limit it to any one particular service. Besides, for management plane, a single SP is needed anyway.

Reply:
We don't currently need different SPs but we do need different RGs and scoping everything just gives us more consistency and flexibility in the future.